src/NUnitFramework/framework/Attributes/PlatformAttribute.cs

```diff
@@ -67,7 +67,7 @@ public void ApplyToTest(Test test)
                 catch (InvalidPlatformException ex)
                 {
                     test.RunState = RunState.NotRunnable;
-                    test.Properties.Add(PropertyNames.SkipReason, ex.Message);
+                    test.Properties.Add(PropertyNames.SkipReason, ex.GetMessageWithoutThrowing());
```

Comment:
Discuss: Should we use this technique for our own thrown exceptions? Or should we trust our own code and use this for outside exceptions? Which approach will make it easier to catch problems? Which approach is clearer? My reaction here was to go and try to figure out why you thought this might throw an exception, which is of course a waste of time.

Reply:
I didn't think about this, but of course this wouldn't throw an exception. It's not as though we were accessing ex.StackTrace here which might throw. I'll undo instances of GetMessageWithoutThrowing where the exception could not have been thrown by user code.

Reply:
I would lean towards handling all the exceptions the same, as this would also work in the future if exceptions are changed. I see it similar to how we often have guards in the beginning of methods even though we now can be sure that the guards are satisfied by examining all call-sites.
But I can see Charlies' point that it can confuse the reader, if he/she tries to figure out why we think that the code might throw an exception.

Reply:
Something like GetMessageFromUserException might give a clearer signal.

Reply:
@CharliePoole I like the way you think. The reason I didn't go with a name like that is that I like the concept that the extension methods are divorced from the methodology they are used in. The name would indicate that they don't work just as well for non-user exceptions and that feels to me like it a confusing message.

Reply:
@mikkelbu Yes. After Charlie pointed it out, I also personally preferred to switch back to `.Message` so that the code isn't misleading (which maybe you saw). Just to confirm, sounds like you're happy with that for now?

Reply:
Yes. I'm happy with that :).